File: algorithms/2_medium/H/00198_house_robber/dp.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        prev, prev_prev = (0, 0)

        for num in nums:
            curr = num + prev_prev
            prev_prev = max(prev_prev, prev)
            prev = curr

        return max(prev, prev_prev)

File: algorithms/2_medium/H/00198_house_robber/dp.rs
impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        let (mut prev, mut prev_prev) = (0_i32, 0_i32);

        for num in nums {
            let curr = num + prev_prev;
            prev_prev = prev_prev.max(prev);
            prev = curr;
        }

        prev.max(prev_prev)
    }
}
